tools/list-repo-maintenance-commits.sh

```diff
@@ -91,23 +93,37 @@ function seen {
 }
 
 # Figure out which dates to check.
-read -r DS DE < <(php -r '
-	$t = new DateTimeImmutable( "00:00:00 UTC" );
-	$d = $t->format( "N" ) - 1;
-	printf(
-		"%s %s\n",
-		$t->modify( - ( $d + 7 ) . " days" )->format( "Y-m-d" ),
-		$t->modify( -$d . " days" )->format( "Y-m-d" )
-	);
-')
+if [[ $1 =~ ^[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]$ ]]; then
+	DS=$1
+	if [[ $2 =~ ^[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]$ ]]; then
+		DE=$2
```

Comment:
Noting that technically this matches "dates" like 1924-90-90. I'd probably use `date` to validate things.